src/libs/actions/PersonalDetails.js

```diff
@@ -87,20 +88,20 @@ function formatPersonalDetails(personalDetailsList) {
 
     // This method needs to be SUPER PERFORMANT because it can be called with a massive list of logins depending on the policies that someone belongs to
     // eslint-disable-next-line rulesdir/prefer-underscore-method
-    Object.keys(personalDetailsList).forEach((login) => {
-        const personalDetailsResponse = personalDetailsList[login];
+    Object.entries(personalDetailsList).forEach(([login, details]) => {
```

Comment:
The terrifying comment above this line seems to imply that `Object.keys()` + `forEach()` is more performant than underscore. I can't really think of why that would be, but at the same time my default is to trust whoever wrote it. 

Since we are changing this to `Object.entries()` is there some way we can verify there are no performance regressions. Maybe testing against production with an account that will have many personal details (someone who has been at Expensify for a while perhaps)?

Reply:
> Since we are changing this to Object.entries() is there some way we can verify there are no performance regressions. Maybe testing against production with an account that will have many personal details (someone who has been at Expensify for a while perhaps)?

Yes, I can add QA steps here.

FWIW [here is the PR](https://github.com/Expensify/App/pull/7649) that introduced that, and the problem with the code before wasn't necessarily that we were using underscore instead of vanilla JS, just that we were using the spread operator in a `reduce`, which is an `O(n^2)` operation because it copies the whole object for every iteration.

Reply:
So it would probably be safe to just use `_.each(personalDetailsList, (details, login) => {...})` because it doesn't have that spread inside a reduce.

Reply:
Added QA steps. Applause has access to some high-volume accounts with large policies (Expensify.org policies I think) that will give them thousands of `personalDetails`.

Reply:
A bit of extra context here: https://github.com/Expensify/App/pull/7649#discussion_r802838523

Reply:
Ok cool. It's a weird comment then and sounds like we could update this to use underscore, but NAB.